scene/gui/line_edit.cpp

```diff
@@ -45,6 +45,84 @@
 #include "editor/editor_settings.h"
 #endif
 
+void LineEdit::grab_focus_edit(bool p_select_all) {
+	if (!is_inside_tree()) {
+		return;
+	}
+	set_focus_edit(true);
+
+	if (p_select_all) {
+		select_all();
+		set_caret_column(text.length());
```

Comment:
Doesn't `select_all()` already set the caret?

Reply:
i think it sets the caret to the start of the selection by default, since pressing the right arrow will move the caret to the end of the selection and remove the selection, and I got used to the old behavior so it may be confusing to change it.